rskj-core/src/main/java/org/ethereum/datasource/DataSourceWithCache.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.datasource;
+
+import co.rsk.util.MaxSizeHashMap;
+import org.ethereum.db.ByteArrayWrapper;
+import org.ethereum.util.ByteUtil;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class DataSourceWithCache implements KeyValueDataSource {
+    private final KeyValueDataSource base;
+    private final Map<ByteArrayWrapper, byte[]> uncommittedCacheMap;
+    private final Map<ByteArrayWrapper, byte[]> committedCacheMap;
+
+    // During the processing of a Raul's fully filled blockchain, the cache
+    // is generating the following hits per block (average)
+    // uncommitedCacheHits = 1340
+    // commitedCacheHits = 248
+    // Processing 134 blocks grows the commitedCache to 100K entries, or approximately
+    // or 25 Mbytes. A cache of about 100 Mbytes seems rasonable. Anyway, for
+    // precaution we'll set the limit to 100K entries.
```

Comment:
This comment is outdated

Reply:
It's not outdated, it's just in the wrong place. I'll move it to the proper place so it gets sense in the right context. ty!